modules/gdscript/gdscript_compiler.cpp

```diff
@@ -278,6 +278,11 @@ int GDScriptCompiler::_parse_expression(CodeGen &codegen, const GDScriptParser::
 				return idx | (GDScriptFunction::ADDR_TYPE_GLOBAL << GDScriptFunction::ADDR_BITS); //argument (stack root)
 			}
 
+			if (codegen.script->tool) {
```

Comment:
Can you make this check stricter by checking if the singleton is defined in ProjectSettings?